src/plugins/credential_management/public/components/utils.ts

```diff
@@ -53,7 +53,7 @@ export async function deleteCredentials(
   selectedCredentials: CredentialsTableItem[]
 ) {
   // TODO: Refactor it
-  selectedCredentials.forEach(function (selectedCredential) {
-    savedObjectsClient.delete('credential', selectedCredential.id);
-  });
+  for (const credential of selectedCredentials) {
+    await savedObjectsClient.delete('credential', credential.id);
+  }
```

Comment:
for poc it is ok, but for production, let's make a bulk api instead of make batch api call like DOS attack

Reply:
Unfortunately, there is no bulk API for deletion - https://www.elastic.co/guide/en/kibana/current/saved-objects-api-delete.html

We can add one, but it is unlikely to be a priority.